src/styleSheetSerializer.js

```diff
@@ -96,6 +96,9 @@ const stripUnreferencedClassNames = (result, classNames) =>
     classNames
       .reduce((acc, className) => acc.replace(new RegExp(`${className}\\s?`,'g'), ''), result);
 
+const stripUnusedStyledComponentRules = result =>
+  result.replace(/[^\n\r]*\.sc-[\w\d]{4,8}\s*{[^}]+}[\n\r]+/, '')
```

Comment:
Not sure why, but during testing, the styled component id's length ranged between 5 and 6 chars. To be safe I set this to between 4 and 8, but it might be good to dial this in to the actual min and max length of a styled component id.